src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/SSRPHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.SNI;
+
+namespace Microsoft.Data.SqlClientX.Handlers.Connection
+{
+    /// <summary>
+    /// Handler to resolve SSRP related ports,instance name etc.
+    /// </summary>
+    internal class SSRPHandler : ContextHandler<ConnectionHandlerContext>
+    {
+        private const int DefaultPort = 1434;
+        public override async ValueTask Handle(ConnectionHandlerContext request, bool isAsync, CancellationToken ct) 
+        {
+            bool isAdmin = request.DataSource.ResolvedProtocol == DataSource.Protocol.Admin;
+            bool instanceNameProvided = !string.IsNullOrWhiteSpace(request.DataSource.InstanceName);
+            bool isPortProvided = request.DataSource.Port == -1;
+
+            if (isAdmin)
+            {
+                if (instanceNameProvided)
+                {
+                    //TODO: resolve DAC port via SSRP
+                    throw new NotImplementedException();
+                }
+                else if (isPortProvided)
+                {
+                    request.DataSource.ResolvedPort = request.DataSource.Port;
+                }
+                else
+                {
+                    request.DataSource.ResolvedPort = DefaultPort;
```

Comment:
Something is a bit off here. The data source port is the one where the Sql Server is running. 
DefaultPort is 1434. That is where the browser runs. 
I am not sure why we are assigning 1434 to the ResolvedPort, since SQL doesn't listen to connections on Default port. Am I missing something ?

Reply:
I read a bit more. The default DAC port is 1434 over TCP and the default Browser Port is 1434 UDP.
So I think I understand your intention now. 
You should rename the default port to DefaultDacPort.